### PR TITLE
[WIP] refactor react card for stream pages

### PR DESCRIPTION
**Pull request by andygnewman**
- [x] use string interpolation
- [x] refactor open / close button logic
- [x] abstract out inline article react class from stream card
- [x] enable appending to dom using bare fragment option from article & styling
- [x] flag for iframe vs fetch
- [x] get o-share bower component working server side
- [x] get all usual article functions working (eg. share, print, follow links in article)

### CODE REVIEW

**Comment by matthew-andrews** [comment]
ha, nice, you're quite right.  probably a lot can be tidied up in this case.

**Comment by matthew-andrews** [comment]
@andygnewman I think this works well enough now — printing (albeit a bit weird), sharing, saving…

File: server/controllers/v2-taxonomy.js [hunk 1]
```diff
@@ -31,7 +31,8 @@ module.exports = function(req, res, next) {
 				items: results[0]
 					.map(articleModel({
 							hasActions: config.hasActions,
-							hasOpen: config.hasOpenInStream
+							hasOpen: config.hasOpenInStream,
+							useDomInsertion: res.locals.flags.streamOpenArticleDOM && isFastFt
 						})),
```
Comment (andygnewman) [inline server/controllers/v2-taxonomy.js]: replace isFastFT with config.hasOpenInStream

**Comment by andygnewman** [comment]
The print one will look a bit weird - probably need to do a scope on the print function to just print the article elements of the DOM.

**Comment by matthew-andrews** [comment]
Hmm…  I think this is also missing galleries and videos…

**Comment by andygnewman** [comment]
Ah yes, galleries get stuck on through client side, are you seeing the placeholders come through?

**Comment by matthew-andrews** [comment]
No, but I'm wondering whether it would be better to take out the duplicated CSS/JS from article for now and bring it back when we figure how we're going to share the code.

**Comment by andygnewman** [comment]
Yes, would be a good idea, as it is only for internal proof of concept at the moment.

**Pull request merged**